src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentQueue.cs

```diff
@@ -255,8 +255,12 @@ public bool IsEmpty
         /// cref="ConcurrentQueue{T}"/>.</returns>
         public T[] ToArray()
         {
-            return ToList().ToArray();
+            List<T> list = ToList();
```

Comment:
It may make sense to call GetHeadTailPositions directly and check if queue is (was, really) empty and return empty. We can assume that ToArray will be typically called in quiescent state, so GetHeadTailPositions should be fairly cheap.

If this same optimization is applied to List<T>.ToArray(), then we do not really need to check the list for emptiness.